sql/handler.h

```diff
@@ -1382,6 +1386,14 @@ struct handlerton
    */
    int (*discover_table_structure)(handlerton *hton, THD* thd,
                                    TABLE_SHARE *share, HA_CREATE_INFO *info);
+
+   /*
+     System Versioning
+   */
+   /** Fill TRT record for update.
+    @param[out]   trt       TRT table which record[0] will be filled with
+                            transaction data. */
+   void (*vers_get_trt_data)(TR_table &trt);
```

Comment:
All I know is that Oracle’s MySQL team is following the [Google C++ style guide, which prohibits non-`const` reference parameters](https://google.github.io/styleguide/cppguide.html#Reference_Arguments). There is only one reply in the thread that you mention, by @vuvova, seemingly to agreeing with the Google C++ style guide: "in" parameters can be (`const`) references, and "out" parameters should be pointers.

I’d also follow the [C++ Core Guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines) whenever applicable, but I guess we cannot fully do so before moving to C++11 or later.

Personally, I have no problem with non-`const` reference parameters. But I think that the coding style should be consistent at least in the subsystem interfaces.

Reply:
This is pure practical: if it is a reference there are not questions that it can be NULL. I would leave it as is if you don't mind.

Reply:
@dr-m is there some official coding standard in MariaDB like Google style guide? Have you heard about `clang-format` tool?

Reply:
@kevgs these are questions to @vuvova.
I would welcome the use of `clang-format` for new code; maybe even some pre-push trigger that would reject code that is not formatted correctly. But I fear that it might be impractical to convert old code. It would complicate code merges, both between MariaDB branches (unless we reformat code in all active branches) and from external sources.

Reply:
@dr-m @vuvova  Then maybe you'll add `.clang-format` files to root directory and `storage/innodb` and make it official? That would be a pleasure to use. Here is what I have for `InnoDB`:
```yaml
UseTab: Always
TabWidth: 8
IndentWidth: 8
BreakBeforeBinaryOperators: NonAssignment
PointerAlignment: Left
AlwaysBreakAfterReturnType: TopLevel
```
And for the rest of code:
```yaml
BreakBeforeBraces: Allman
SpaceBeforeAssignmentOperators: false
```
I can open a PR I you want.

That's not fun to format manually in 2017.

Reply:
You are right, it should be automated, and I guess `clang-format` is a more practical choice than to force everyone to use GNU Emacs, whose `cc-mode` I tweaked years ago for InnoDB, like this:
```elisp
	(setq
	 c-basic-offset 8
	 c-label-minimum-indentation 0)
	(add-to-list 'c-offsets-alist '(c . 0))
	(add-to-list 'c-offsets-alist '(arglist-intro . +))
	(add-to-list 'c-offsets-alist '(label . [0]))
```
The `c-label-minimum-indentation` was mandated by Heikki Tuuri. I guess we would want to keep that one to avoid white-space changes to existing code.
@kevgs Can you try to check if all these are covered by the `.clang-format`?